classes/module/Module.php

```diff
@@ -472,6 +477,60 @@ public function install()
         return true;
     }
 
+    protected function checkOverrideConflicts(): bool
+    {
+        if (!is_dir($this->getLocalPath() . 'override')) {
+            return true;
+        }
+
+        foreach (Tools::scandir($this->getLocalPath() . 'override', 'php', '', true) as $file) {
+            $module_override_path = $this->getLocalPath() . 'override/' . $file;
+            $prestashop_override_path = _PS_ROOT_DIR_ . '/override/' . $file;
+
+            if (file_exists($prestashop_override_path)) {
+                if (!$this->compareOverrideFiles($module_override_path, $prestashop_override_path)) {
+                    $this->_errors[] = Context::getContext()->getTranslator()->trans(
+                        'The override file %1$s conflicts with an existing override in %2$s.',
+                        [$module_override_path, $prestashop_override_path],
+                        'Admin.Modules.Notification'
+                    );
+
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    private function compareOverrideFiles(string $module_override_path, string $prestashop_override_path): bool
+    {
+        $module_methods = $this->getClassMethodsFromContent(file_get_contents($module_override_path));
+        $prestashop_methods = $this->getClassMethodsFromContent(file_get_contents($prestashop_override_path));
+
+        foreach ($module_methods as $method) {
+            if (in_array($method, $prestashop_methods)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private function getClassMethodsFromContent(string $content): array
```

Comment:
There must be any explanation how function works. There is missing return types and types.

Reply:
done